pkg/caveats/types/registration.go

```diff
@@ -80,7 +80,7 @@ func registerGenericType(
 }
 
 // registerCustomType registers a custom type that wraps a base CEL type.
-func registerCustomType(keyword string, baseCelType *cel.Type, converter typedValueConverter, opts ...cel.EnvOption) VariableType {
+func registerCustomType[T CustomType](keyword string, baseCelType *cel.Type, converter typedValueConverter, opts ...cel.EnvOption) VariableType {
```

Comment:
I believe this generic is not being used and can be dropped

Reply:
It is explicitly added to type check against the interface, to ensure it meets the `CustomType` requirements